packages/contracts-bedrock/test/L2/SuperchainWETH.t.sol

```diff
@@ -0,0 +1,326 @@
+    function test_sendERC20_sufficientFromCustomGasTokenChain_succeeds() public {
+        // Arrange
+        uint256 amount = 1000;
+        uint256 chainId = 1;
+        vm.deal(alice, amount);
+        vm.prank(alice);
+        superchainWeth.deposit{ value: amount }();
+        vm.mockCall(address(l1Block), abi.encodeCall(l1Block.isCustomGasToken, ()), abi.encode(true));
+
+        // Act
+        vm.expectEmit(address(superchainWeth));
+        emit Transfer(alice, address(0), amount);
+        vm.expectEmit(address(superchainWeth));
+        emit SendERC20(alice, bob, amount, chainId);
+        vm.expectCall(Predeploys.ETH_LIQUIDITY, abi.encodeCall(ETHLiquidity.burn, ()), 0);
+        vm.expectCall(
+            Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER,
+            abi.encodeCall(
+                IL2ToL2CrossDomainMessenger.sendMessage,
+                (chainId, address(superchainWeth), abi.encodeCall(superchainWeth.relayERC20, (bob, amount)))
+            ),
+            1
+        );
+        vm.prank(alice);
+        superchainWeth.sendERC20(bob, amount, chainId);
+
+        // Assert
+        assertEq(alice.balance, 0);
+        assertEq(superchainWeth.balanceOf(alice), 0);
+    }
+
+    /// @notice Tests that the sendERC20 function reverts when called with insufficient balance.
+    function test_sendERC20_insufficientBalance_fails() public {
+        // Arrange
+        uint256 amount = 1000;
+        uint256 chainId = 1;
+        vm.deal(alice, amount);
+        vm.prank(alice);
+        superchainWeth.deposit{ value: amount }();
+
+        // Act
+        vm.expectRevert();
+        superchainWeth.sendERC20(bob, amount + 1, chainId);
+
+        // Assert
+        assertEq(alice.balance, 0);
+        assertEq(superchainWeth.balanceOf(alice), amount);
+    }
+
+    /// @notice Tests that the sendERC20 function always succeeds when called with a sufficient
+    ///         balance no matter the sender, amount, recipient, or chain ID.
+    /// @param _amount The amount of WETH to send.
+    /// @param _caller The address of the caller.
+    /// @param _recipient The address of the recipient.
+    /// @param _chainId The chain ID to send the WETH to.
+    function testFuzz_sendERC20_sufficientBalance_succeeds(
+        uint256 _amount,
+        address _caller,
+        address _recipient,
+        uint256 _chainId
+    )
+        public
+    {
+        // Assume
+        _amount = bound(_amount, 0, type(uint248).max);
+
+        // Arrange
+        vm.deal(_caller, _amount);
+        vm.prank(_caller);
+        superchainWeth.deposit{ value: _amount }();
+
+        // Act
+        vm.expectEmit(address(superchainWeth));
+        emit Transfer(_caller, address(0), _amount);
+        vm.expectEmit(address(superchainWeth));
+        emit SendERC20(_caller, _recipient, _amount, _chainId);
+        vm.expectCall(Predeploys.ETH_LIQUIDITY, abi.encodeCall(ETHLiquidity.burn, ()), 1);
```

Comment:
Why use the overload with 3 args here when its not used elsewhere? I don't think the 2 arg version is an alias for the 3 arg version with 1

https://github.com/foundry-rs/forge-std/blob/c28115db8d90ebffb41953cf83aac63130f4bd40/src/Vm.sol#L1761

Reply:
I think the 3 arg version is better because it asserts the specific number of times that a function is called instead of just asserting that it gets called at all.